starlette/requests.py

```diff
@@ -223,7 +223,7 @@ async def stream(self) -> typing.AsyncGenerator[bytes, None]:
                 body = message.get("body", b"")
                 if body:
                     yield body
-                if not message.get("more_body", False):
+                if self._stream_consumed:
```

Comment:
Does this change is needed? The `_CachedRequest` doesn't change the value of `self._stream_consumed`. 🤔

Reply:
The test `test_read_request_stream_in_dispatch_after_app_calls_body` fails without this logic.

Hmm... Why the `more_body` doesn't matter? Like, not considering the `BaseHTTPMiddleware`, why the `more_body` doesn't matter to exit?

Hmmm... If we receive 2 chunks of body, how this works? It doesn't look like we have a test that covers standalone `Request` with multiple chunks. 🤔

Reply:
Or am I missing something?

Reply:
I don't recall how to do it from the TestClient's POV, but I thought about sending a stream with 2 chunks. Maybe you can use httpx directly if you can't do it with the TestClient. 

I guess that would be enough to break this logic here, since the value of stream_consumed will not change

Reply:
Ok yes, you were right, I did have a bug, good catch. I still need to modify `Request` a bit, I added a couple of tests to explain why. TLDR is we were marking the stream as consumed as soon as you call `stream()` but in reality you can call stream, get one message and then _call steam again_ before it is consumed. Let me know if it's clear now.